### PR TITLE
Updating SupportButton title ui characteristics

### PR DESCRIPTION
**Pull request by suniljose25**
## Hey, I just made a Pull Request!

When the title prop is provided for `SupportButton` it is rendered as a `MenuItem` which by default has the ui characteristics of a Button(has cursor as `pointer`, has click animation).

This change passes `false` as value to the button prop of `MenuItem`, which prevents the button UI characteristics from being included.

Screenshots:
Before:

https://github.com/backstage/backstage/assets/14274064/8b0d72a1-7079-428d-99a6-449d96596d9a

After:

https://github.com/backstage/backstage/assets/14274064/906e116c-0e3b-49fc-a323-8edcbe7b4dea

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — CHANGES_REQUESTED** [review]
Hi @suniljose25, is it possible to attach the screenshot of the proposed changes? It makes the review easier 🙏

**Comment by suniljose25** [comment]
> Hi @suniljose25, is it possible to attach the screenshot of the proposed changes? It makes the review easier 🙏

I have added the before and after screen recordings.

This change that we see in the recording is only effective when the text is passed as the title prop.

In all of Backstage, the text is only passed to the SupportButton as children. I made a change to pass it as text in local specifically to record the effect.

**Comment by suniljose25** [comment]
Also @vinzscam any ideas why the E2E tests are failing? How do we resolve them?

File: .changeset/twenty-masks-dance.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-components': patch
+---
+
+Fixed bug in SupportButton where the title was rendered with the characteristics of a button.
```
Comment (vinzscam) [inline .changeset/twenty-masks-dance.md:5]:
```suggestion
Fixed a bug in `SupportButton` where the title was rendered with the characteristics of a button.
```

**Review by vinzscam — APPROVED** [review]
oh amazing thank you!

**Pull request merged**